### PR TITLE
Week 3 release

### PR DESCRIPTION
**Pull request by alexdarie**
Client options (--register, --find-all, --get, --delete, --update) are fully functional. I'm working on the book options: delete or update a book. Also implemented a hack to restart our application by typing "restart". It waits for 3 seconds to add some drama, then it reloads the files.

### CODE REVIEW

File: src/main/java/Main.java [hunk 1]
```diff
@@ -33,10 +36,19 @@ public static void main (String[] args) {
...
+            BufferedReader keyboard = new BufferedReader(new InputStreamReader(System.in));
+            ConsoleApplication consoleApplication = new ConsoleApplication(bookService, clientService, keyboard);
+            int status = consoleApplication.run();
+            while( status == 7 ) {
+                bookRepository.loadFile();
+                authorRepository.loadFile();
+                clientRepository.loadFile();
+                System.out.println("Session restarted..");
+                status = consoleApplication.run();
+            }
```
Comment (alexdarie) [inline src/main/java/Main.java]: hack to refresh the files

File: src/main/java/Repository/FileRepository.java [hunk 2]
```diff
@@ -107,4 +107,29 @@ private void saveToFile(T entity) throws RepositoryException {
...
+        try ( BufferedWriter bw = Files.newBufferedWriter(path) ) {
+            for(T e: super.findAll()) {
+                bw.write(e.toTXF());
+                bw.newLine();
+            }
+        } catch (IOException exc) {
+            throw new RepositoryException(exc.getMessage());
+        }
+    }
+
```
Comment (alexdarie) [inline src/main/java/Repository/FileRepository.java]: Implemented such that now all the changes will be persistent in a text file. Update and delete rewrite the files.. 👎

File: src/main/java/Repository/Repository.java [hunk 3]
```diff
@@ -71,27 +71,16 @@ public Repository(ValidatorInterface<T> validator) {
...
-        this.validateOrThrowException(entity);
-        if (null == this.elements.get(entity.getId())) {
-            elements.put(entity.getId(), entity);
-            return Optional.empty();
+        if (null != entity) {
+            this.validateOrThrowException(entity);
+            if (null == this.elements.get(entity.getId())) {
+                elements.put(entity.getId(), entity);
+                return Optional.empty();
+            }
```
Comment (alexdarie) [inline src/main/java/Repository/Repository.java]: Got rid of the previous implementation cuz' it sucked

File: src/main/java/UI/ConsoleApplication.java [hunk 4]
```diff
@@ -38,9 +40,10 @@
      * Initializes the bookService and the predefined data.
      * @param bookService
      */
-    public ConsoleApplication(BookService bookService, ClientService clientService) {
+    public ConsoleApplication(BookService bookService, ClientService clientService, BufferedReader keyboard) {
         this.bookService = bookService;
         this.clientService = clientService;
+        this.keyboard = keyboard;
```
Comment (alexdarie) [inline src/main/java/UI/ConsoleApplication.java]: related to the restart hack

File: src/test/java/Repository/RepositoryTest.java [hunk 5]
```diff
@@ -229,7 +236,7 @@ void update() {
                 "magpi.com",
                 "Paperback",
                 340,
-                new LinkedList(Arrays.asList(new Author("Issue", "64"))),
+                new LinkedList(Arrays.asList(1)),
```
Comment (alexdarie) [inline src/test/java/Repository/RepositoryTest.java]: removed some useless things

**Comment by alexdarie** [comment]
All done on the book side

File: src/main/java/Main.java (hunk 1 above)
Comment (redount2k9) [reply src/main/java/Main.java]: bv tata mare hacker

**Pull request merged**